### ISSUE TITLE
Export is broken in Bitwarden Web Vault 2.0

### ISSUE THREAD

**Issue by setyb** [bitwarden/web#230]
I noticed that the size of the export file has been shrinking even though I have added data.

I carefully inspected the export CSV file, and sure enough, numerous Bitwarden login items are missing from the export file that used to get properly exported.

I have confirmed the logins exist in Bitwarden, are shown in the Web Vault, but are not exported.

I performed the export multiple times to make sure it wasn't a one-time issue.  It's not.  The bug is reproducible every time.

Something is seriously broken with export right now.  It's not exporting all the logins stored in Bitwarden.  (And of course, it still doesn't export cards or identities.)

**Comment by kspearrin** [comment]
Are these items in an organization? Organization items are no longer included in personal vault exports. You would need to export from the org.

**Comment by setyb** [comment]
Hi Kyle.  Yes, the items were in an organization.  I didn't go through the missing items one by one to make sure they were all in an organization, but the first 3 were, so that seems like a likely explanation.

If that's the case, it sounds like *(a fairly serious)* UX issue.  When the user sees items on the screen, and then selects *Export*, they will assume (like I did) that all the items will be exported.

A good solution will be to present a dialog box that asks the user if they want to export ***all*** of the items displayed or only those not in an organization.

**Comment by kspearrin** [comment]
We can update the text to inform the user that organization items are not included. I don't intend to make it available as an option. Organization exports are done from the org admin area.

**Comment by setyb** [comment]
It's an interesting dilemma.  All my items in an organization originated from within a login (non-organization).  So it's strange to add something to an organization and then have it no longer part of it's origin.

**Comment by kspearrin** [comment]
When you share an item to an organization you no longer own that item. The organization now owns it.

**Comment by setyb** [comment]
Although that may be true in Bitwarden, the user has no way of knowing it.  It's not obvious or typical.  Outside of Bitwarden, ***sharing*** never transfers ownership.  ***Sharing*** always shares data from party A with party B and keeps ownership with party A.

It also doesn't work that way in the real world.  For example, if you shared your car with me, I don't get ownership.

In the UX world, we call these things *mental models*, and Bitwarden is violating the established *mental model*.

**Comment by kspearrin** [comment]
We try to make that clear within the share dialog message:

![image](https://user-images.githubusercontent.com/1190944/43407639-45fb551c-93ed-11e8-849f-38ac19819986.png)

While I agree with you, I'm not sure what could be done to improve that other than choose some word other than share perhaps?

**Comment by mprasil** [comment]
I have to agree with @setyb here, the wording is a bit confusing. I've seen few people already being confused by the share functionality. (Let's be honest, no one reads the long description :D)

I'd say "Transfer" might be better world to use here, though I'm not native speaker, so take that with grain of salt.

**Comment by setyb** [comment]
I agree that *Transfer* is a better term for the current functionality.  The "traditional" share icon might warrant a change as well.

But, for a moment, lets take a step back and look at the big picture.  Do users really want to transfer an item to an organization or do they want to share it with the organization?  (That's a real question, not a leading one.)  I see value with both viewpoints.

**Comment by mprasil** [comment]
I'd argue that once you share password in real life it's no longer yours. It's basically controlled leaking, when you leak your password the correct action to make it yous again is to create a new one. Bitwarden should reflect that fact (and it does now) by making sure you no longer own it. (so it does not show in exports for example)

From security standpoint it does the right thing functionally to not give you the false illusion of still being in control of your password, but the wording needs to reflect that.

**Comment by Sharky444** [comment]
The word _share_ needs to be replaced, as it's not sharing.

**Comment by DraneLixX** [comment]
`setyb commented on 31 Jul 2018`
> I agree that _Transfer_ is a better term for the current functionality. The "traditional" share icon might warrant a change as well.
> 
> But, for a moment, lets take a step back and look at the big picture. Do users really want to transfer an item to an organization or do they want to share it with the organization? (That's a real question, not a leading one.) I see value with both viewpoints.

I have to agree with setyb first i also thought i exportet everything and then was shocked where everything is after inspecting and rolling back bwdata i had everything back.

Things like that are no nice experience as you can guess but i think for the most of the users it would be more useful if you would not loose the ownership of the data. or if would be asked if you want to also export data from Org. (If you have permission that should be a easy check for rights)

**Comment by setyb** [comment]
This is an interesting conundrum. If I'm understanding everyone's feedback correctly, we all agree that the word "share" does not correlate well with the current functionality.

While thinking about this, I reflected on why a user or admin wants to let another entity have access to private information (such as a password), and what they are really trying to accomplish.

This led to the idea that users and admins generally want one of two things:

1. Transfer/move an item from a person/organization to a different person/organization.

- I think both "transfer" and "move" are good English words to use for this functionality.  The word "transfer" has a useful physical connotation to it, and the word "move" is very common across most operating systems when dealing with moving computer files and folders.  I have a _slight_ preference for "move", but I think either word works well.
- This is the current functional implementation, and requires only changing string resource files and an icon.  At least in English, since "Move" is shorter than "Share" (in every common font), we can be sure that no UI elements or pages will need to be resized.  "Transfer" is a touch longer than "Share", but after looking at (hopefully all) places in the UI where it is used, it also should not require any code modifications.  Translations in other languages could possibly require _minor_ code changes, but it is unlikely based on manual and AI-based translations I performed in 12 common languages.

2. Share/disclose an item to another person/organization without changing ownership.

- In this case, the user retains ownership of the item, but discloses it to another person/organization.  Whether or not that second entity can _modify/edit_ the item is determined by the person sharing the item.
- I suggested that when you share an item (such as a vehicle), you do not transfer ownership.  Miro (@mprasil) suggested that when you share a password, it's no longer yours, and that it's a form of "controlled leaking".  _Both of us are correct._  My interpretation of the word "share" revolves around the most common definition of "share", while Miro's interpretation revolves around the specific reality of sharing passwords.  Since both interpretations are accurate and reasonable, the word "share" may not be ideal for this functionality.
- For this functionality, I haven't come up with a great word. "Disclose", "Reveal", and "Expose" all come to mind, but I don't like any of them more than "Share".  And if the item's owner is allowing other parties to modify the item, "Share" is more accurate.  "Share Jointly" is a little more precise, but cumbersome.  Do others have ideas on a better term?  How about "Allow Access"?
- This functionality requires adding some new code, as bitwarden currently does not have this feature.

I welcome and encourage feedback.

**Comment by Jip-Hop** [comment]
My exports no longer contain all my items since I started using the Sharing feature, which I didn't realize for quite some time... I use the export feature to keep regular backups of my Vault. But because of the incomplete exports this has become inconvenient. There's no way to export shared items from the desktop or mobile (iOS) apps. So now I have to additionally login to the Bitwarden web app to export my shared items from there.

Ideally I'd be able to export "All items" from the app (including shared items). But until then a warning when exporting a vault would be most welcome. Something like: "Shared items will be missing. Please export shared items separately: link to instructions".

**Comment by djsmith85** [comment]
The inital request to make it clearer what is included in the exports (personal vs. organization) should be addressed with: https://github.com/bitwarden/web/pull/1504. The notice is now also shown on other clients.

The Share functionality was also renamed to "Move to organization" a while ago.

@setyb Would you agree that this can be closed? Any further changes would be probably feature-requests anyway.
